### ISSUE TITLE
Task 2.4.3: Implement Mempool for `StateService`

### ISSUE THREAD

**Issue by idky137** [zingolabs/zingo-proxy#118]
- Blocked on #54

**Comment by pacu** [comment]
@idky137 is this blocked by https://github.com/ZcashFoundation/zebra/issues/8447?

**Comment by idky137** [comment]
No, the design of this has changed as we originally thought we would be fetching mempool data from the `ReadStateService` rather than the JsonRPC service. For this reason we will now use the same mempool implementation added here https://github.com/zingolabs/zaino/pull/152. This will be added to Zaino's `StateService` in https://github.com/zingolabs/zaino/issues/68 when we add the rest of the Lightwallet gRPC service implementations.
